example/main.go

```diff
@@ -106,19 +109,51 @@ func Main() int {
 		case errors.Is(err, io.EOF):
 			log.Infof("EOF received, exiting.")
 			return 0
+		case errors.Is(err, terminal.ErrInterrupted):
+			interrupts++
+			if interrupts >= 3 {
+				log.Infof("Triple interrupt, exiting.")
+				return 0
+			}
+			log.Infof("Interrupted, resetting, use exit or Ctrl-d. to exit.")
+			t.ResetInterrupts(context.Background())
 		default:
 			return log.FErrf("Error reading line: %v", err)
 		}
-		log.Infof("Read line got: %q", cmd)
 		// Save previous command validity to know whether this one should replace it in history or not.
 		previousCommandWasValid = isValidCommand
 		isValidCommand = false // not valid unless proven otherwise (reaches the end validations etc)
+		if cmd == "" {
+			// we do get empty reads on interrupt
+			log.LogVf("Empty command read")
+			continue
+		}
+		log.Infof("Read line got: %q", cmd)
 		switch {
 		case cmd == exitCmd:
```

Comment:
I'm unsure to understand this code, I mean the whole interception of command

I mean are the following command caught?

- so a command starting or ending
- `whatever; exit`
- `whatever && exit`
- `whatever || exit`